src/api/app/models/local_build_result.rb

```diff
@@ -0,0 +1,31 @@
+class LocalBuildResult
+  include ActiveModel::Model
+  attr_accessor :project, :package, :repository, :architecture, :code
+
+  def self.find_by(opts)
+    find_by_project_and_package(opts[:project], opts[:package]).select { |buildresult|
+                                                                         buildresult.repository == opts[:repository] &&
+                                                                         buildresult.architecture == opts[:architecture]
+                                                                      }
+  end
+
+  def self.find_by_project_and_package(project, package)
+    buildresults = Buildresult.find_hashed( project: project, package: package, view: 'status', multibuild: '1', locallink: '1')
```

Comment:
Also, this method should be private if we are only going to use it from the `find_by` function.